libraries/c_sdk/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -1230,7 +1281,56 @@ IotMqttError_t IotMqtt_Connect( const IotMqttNetworkInfo_t * pNetworkInfo,
     {
         IotLogInfo( "New MQTT connection %p established.", pMqttConnection );
         /* Set the output parameter. */
-        *pMqttConnection = pNewMqttConnection;
+        *pMqttConnection = newMqttConnection;
+
+        /* Fill in time utility function pointer. */
+        callbacks.getTime = getTimeMs;
+
+        /* Getting the free index from the MQTT connection to MQTT context mapping array. */
+        contextIndex = _IotMqtt_getFreeIndexFromContextConnectionArray();
+
+        if( contextIndex < 0 )
+        {
+            IotLogError( "(MQTT connection %p) Failed to allocate memory for "
+                         "the MQTT context and the MQTT Connection Mapping. Update the MAX_NO_OF_MQTT_CONNECTIONS"
+                         "config value to resolve the error. ",
+                         newMqttConnection );
+            IOT_SET_AND_GOTO_CLEANUP( IOT_MQTT_NO_MEMORY );
+        }
+
+        /* Creating Mutex for the synchronization of MQTT Context used for sending the packets
+         * on the network using MQTT LTS API. */
+        bool contextMutex = IotMutex_Create( &( connToContext[ contextIndex ].contextMutex ), true );
+
+        if( contextMutex == true )
+        {
+            /* Assigning the MQTT Connection. */
+            connToContext[ contextIndex ].mqttConnection = newMqttConnection;
+
+            /* Assigning the Network Context to be used by this MQTT Context. */
+            connToContext[ contextIndex ].networkContext.networkConnection = pNetworkConnection;
```

Comment:
wouldn't networkConnection also need to be pNetworkConnection as it is a pointer?